lib/lib-table-cluster/src/test/java/org/sagebionetworks/table/cluster/TableIndexDAOImplUnitTest.java

```diff
@@ -0,0 +1,218 @@
+package org.sagebionetworks.table.cluster;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Sets;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.sagebionetworks.repo.model.ObjectType;
+import org.sagebionetworks.repo.model.table.ColumnModel;
+import org.sagebionetworks.repo.model.table.ColumnType;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.datasource.DataSourceTransactionManager;
+import org.springframework.transaction.support.TransactionTemplate;
+
+@ExtendWith(MockitoExtension.class)
+public class TableIndexDAOImplUnitTest {
+	@Mock
+	private DataSourceTransactionManager mockTransactionManager;
+	@Mock
+	private TransactionTemplate mockWriteTransactionTemplate;
+	@Mock
+	private TransactionTemplate mockReadTransactionTemplate;
+	@Mock
+	private JdbcTemplate mockTemplate;
+	@Mock
+	private NamedParameterJdbcTemplate mockNamedTemplate;
+
+	@Spy
+	@InjectMocks
+	TableIndexDAOImpl spyDao;
+
+
+
+	@Test
+	public void testValidateMaxListLengthInAnnotationReplication_noListColumns(){
+
+		ColumnModel bar = new ColumnModel();
+		bar.setId("1234");
+		bar.setName("bar");
+		bar.setColumnType(ColumnType.INTEGER);
+
+		List<ColumnModel> currentSchema = Arrays.asList(bar);
+
+
+		ObjectType objectType = ObjectType.FILE;
+		long viewId = 123L;
+		long viewTypeMask = 1L;
+		Set<Long> allContainersInScope = Sets.newHashSet(111L,222L);
+		Set<Long> objectIdFilter = null;
+
+		// method under test
+		spyDao.validateMaxListLengthInAnnotationReplication(objectType,viewId,viewTypeMask,
```

Comment:
You might want to check that there are no DB interactions instead and wrap it into the assertDoesNotThrow to give it context